### PR TITLE
[ADF-1868] Update to Angular 5

### PR DESCRIPTION
**Pull request by eromano**
**Please check if the PR fulfills these requirements**

> - [x] The commit message follows our [guidelines](https://github.com/Alfresco/alfresco-ng2-components/wiki/Commit-format)
> - [x] Tests for the changes have been added (for bug fixes / features)
> - [x] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x")

> - [ ] Bugfix
> - [ ] Feature
> - [ ] Code style update (formatting, local variables)
> - [x] Refactoring (no functional changes, no api changes)
> - [ ] Build related changes
> - [ ] Documentation
> - [ ] Other... Please describe:

**What is the current behaviour?** (You can also link to an open issue here)
https://issues.alfresco.com/jira/browse/ADF-1868
https://issues.alfresco.com/jira/browse/ADF-1869
https://issues.alfresco.com/jira/browse/ADF-1870

**What is the new behaviour?**

**Does this PR introduce a breaking change?** (check one with "x")

> - [ ] Yes
> - [ ] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

**Comment by eromano** [comment]
@popovicsandras all seems to work only the test for the breadcrumb node selector Ia not able to make them works 2  ...documentlist/src/components/content-node-selector/content-node-selector.component.spec.ts maybe you knows it better

File: demo-shell-ng2/package.json [hunk 1]
```diff
@@ -70,7 +72,7 @@
     "@angular/platform-browser-dynamic": "5.0.0",
     "@angular/router": "5.0.0",
     "@ngx-translate/core": "8.0.0",
-    "alfresco-js-api": "1.10.0-beta5",
+    "alfresco-js-api": "^1.10.0-79d7c47d02f9910e657fea48a4f7d4d37a218959",
```
Comment (popovicsandras) [inline demo-shell-ng2/package.json]: I think this was by mistake, wasn't it? :P

File: demo-shell-ng2/package.json [hunk 2]
```diff
@@ -6,18 +6,20 @@
   "scripts": {
     "ng": "ng",
     "start": "npm run server-versions && rimraf dist && ng serve --host 0.0.0.0 --app=0 --open --aot=true",
-    "start:dev": "npm run clean-ng2-component-angular && npm run server-versions && rimraf dist && ng serve --host 0.0.0.0 --app=1 pp-dev --proxy-config proxy.conf.js --open",
+    "start:dev": "npm run style:dev & npm run clean-ng2-component-angular && npm run server-versions && rimraf dist && ng serve --host 0.0.0.0 --app=1 pp-dev --proxy-config proxy.conf.js --open",
     "start:dist": "npm run server-versions && rimraf dist && ng serve --host 0.0.0.0 --disable-host-check --aot=false --prod --app=0",
     "build": "npm run server-versions && rimraf dist && ng build  -app=0",
-    "build:dev": "npm run server-versions && rimraf dist && ng build -app=1",
+    "build:dev": "npm run style:dev & npm run server-versions && rimraf dist && ng build -app=1",
+    "style:dev": "npm run webpack -- --config  config/webpack.style.js --progress --profile --bail --watch",
```
Comment (popovicsandras) [inline demo-shell-ng2/package.json]: Question:
What about production builds?

Comment (eromano) [reply demo-shell-ng2/package.json]: the production build use the style already builded in the node_modules folder you don't need to rebuild it

File: demo-shell-ng2/src/tsconfig.dev.json [hunk 3]
```diff
@@ -5,7 +5,7 @@
     "module": "es2015",
     "rootDir": "..",
     "baseUrl": ".",
-    "skipLibCheck": true,
+    "skipLibCheck": false,
```
Comment (DenisVuyka) [inline demo-shell-ng2/src/tsconfig.dev.json]: we can remove this property. The default value is "false" and usually it is not present in the tsconfig file unless you want to set it to "true"

**Pull request merged**